subprojects/version-control/src/main/java/org/gradle/vcs/git/internal/GitVersionControlSystem.java

```diff
@@ -194,16 +188,24 @@ private static GradleException wrapGitCommandException(String commandName, URI r
     }
 
     private static class DefaultTransportConfigCallback implements TransportConfigCallback {
+
         @Override
         public void configure(Transport transport) {
             if (transport instanceof SshTransport) {
                 SshTransport sshTransport = (SshTransport) transport;
-                sshTransport.setSshSessionFactory(new JschConfigSessionFactory() {
-                    @Override
-                    protected void configure(OpenSshConfig.Host hc, Session session) {
-                        // TODO: This is where the password information would go
-                    }
-                });
+
+                SshdSessionFactory factory = new SshdSessionFactory(
+                    new JGitKeyCache(),
+                    new DefaultProxyDataFactory()
+                );
+
+                try {
+                    Runtime.getRuntime().addShutdownHook(new Thread(factory::close));
+                } catch (IllegalStateException e) {
+                    // ignore - the VM is already shutting down
+                }
+
+                sshTransport.setSshSessionFactory(factory);
```

Comment:
It seems this is how to replace jsch

https://www.eclipse.org/forums/index.php?t=msg&th=1106995&goto=1838158&#msg_1838158

https://git.eclipse.org/r/plugins/gitiles/jgit/jgit/+/refs/heads/master/org.eclipse.jgit.pgm/src/org/eclipse/jgit/pgm/TextBuiltin.java#219